src/upload/form/project_assets.rs

```diff
@@ -73,3 +73,94 @@ impl ProjectAssets {
         self.script_path.clone()
     }
 }
+
+pub struct Module {
+    pub path: PathBuf,
+    pub module_type: ModuleType,
+}
+
+impl Module {
+    pub fn new(path: PathBuf) -> Result<Module, failure::Error> {
+        let extension = path
+            .extension()
+            .ok_or(failure::err_msg(
+                "extension required to determine module type",
+            ))?
+            .to_string_lossy();
+
+        let module_type = match extension.as_ref() {
+            "mjs" => ModuleType::ES6,
+            "js" => ModuleType::CommonJS,
+            "wasm" => ModuleType::Wasm,
```

Comment:
does it mean that wasm/text are considered as JavaScript Module?

Reply:
As in they can be imported like a normal javascript file could be imported, yes

Reply:
Wasm and Text are not standard JavaScript Module, I don't think we should do that. I'm concern that our implementation will break once they are standardized

Reply:
Wasm modules have been in the standardization process for years. We shouldn't bind our hands waiting for them. We're going to do our thing, and if it turns out something different gets standardized, we can easily add support for the standard later, via versioning or something.

For now, as implemented in the runtime, importing a Wasm file returns a compiled-but-not-instantiated Wasm module, and importing a text file produces a string of the contents.

Reply:
Importing a text file sounds trivial enough, i'm not really concerned about that. Wasm however, has been in discussion for years because the integration with JavaScript Module or support for Wasi isn't easy.  compiled-but-not-instantiated is against the spec in its current form.

There's also JSON module that are in the standardisation process that we could add.

Reply:
We have to support Wasm, so we have to do something off-spec for now. Compiled-but-not-instantiated is the easiest, most straightforward thing, and should allow existing Wasm bindings generators to work with minimal changes.